### ISSUE TITLE
loadPartitionedParquetAlignments fails with Reference.all

### ISSUE THREAD

**Issue by heuermh** [bigdatagenomics/adam#1967]
```
scala> import org.bdgenomics.adam.rdd.ADAMContext._
import org.bdgenomics.adam.rdd.ADAMContext._

scala> val alignments = sc.loadAlignments("adam-core/src/test/resources/small.sam")
alignments: org.bdgenomics.adam.rdd.read.AlignmentRecordRDD = RDDBoundAlignmentRecordRDD with 2 reference sequences, 0 read groups, and 2 processing steps

scala> alignments.saveAsPartitionedParquet("small.partitioned.alignments.adam")

scala> val partitioned = sc.loadPartitionedParquetAlignments("small.partitioned.alignments.adam")
partitioned: org.bdgenomics.adam.rdd.read.AlignmentRecordRDD = DatasetBoundAlignmentRecordRDD with 2 reference sequences, 0 read groups, and 2 processing steps

scala> partitioned.sequences
res1: org.bdgenomics.adam.models.SequenceDictionary =
SequenceDictionary{
1->249250621, 0
2->243199373, 1}

scala> partitioned.dataset.count()
res0: Long = 20

scala> partitioned.dataset.filter(partitioned.dataset.col("contigName").equalTo("1")).count()
res2: Long = 20

scala> partitioned.dataset.filter(partitioned.dataset.col("contigName").equalTo("2")).count()
res3: Long = 0

scala> val chr1 = sc.loadPartitionedParquetAlignments("small.partitioned.alignments.adam", Seq(ReferenceRegion.all("1")))
chr1: org.bdgenomics.adam.rdd.read.AlignmentRecordRDD = DatasetBoundAlignmentRecordRDD with 2 reference sequences, 0 read groups, and 2 processing steps

scala> chr1.dataset.count()
res4: Long = 0

scala> val chr1 = sc.loadPartitionedParquetAlignments("small.partitioned.alignments.adam", Seq(ReferenceRegion.fromGenomicRange("1", 0L, 249250621L)))
chr1: org.bdgenomics.adam.rdd.read.AlignmentRecordRDD = DatasetBoundAlignmentRecordRDD with 2 reference sequences, 0 read groups, and 2 processing steps

scala> chr1.dataset.count()
res5: Long = 20
```

Ping @jpdna for triage

**Comment by jpdna** [comment]
Is it expected that:
```
scala> ReferenceRegion.all("1")
res7: org.bdgenomics.adam.models.ReferenceRegion = ReferenceRegion(1,0,9223372036854775807,INDEPENDENT)
```
surely 9223372036854775807 is not correct?

If I try to use that ReferenceRegion directly I get an error printed to the console
```
scala> val chr1 = sc.loadPartitionedParquetAlignments("small.partitioned.alignments.adam", Seq(ReferenceRegion(1,0,9223372036854775807,INDEPENDENT) )
<console>:1: error: integer number too large
val chr1 = sc.loadPartitionedParquetAlignments("small.partitioned.alignments.adam", Seq(ReferenceRegion(1,0,9223372036854775807,INDEPENDENT) )
```
but it would seem that error is getting swallowed by the loadPartitionedParquetAlignments call.

**Comment by jpdna** [comment]
Ah yeah 9223372036854775807 is Long.MaxValue at 
https://github.com/bigdatagenomics/adam/blob/1bfddc69a86212ffa6810db79fd30da178ea30a4/adam-core/src/main/scala/org/bdgenomics/adam/models/ReferenceRegion.scala#L165
I'm looking further...

**Comment by fnothaft** [comment]
Closed by #1980.
